engine/src/main/java/org/terasology/engine/subsystem/lwjgl/Lwjgl2Sync.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine.subsystem.lwjgl;
+
+import org.lwjgl.glfw.GLFW;
+
+public final class Lwjgl2Sync {
+
+    private Lwjgl2Sync() {
+    }
+
+    /**
+     * number of nano seconds in a second
+     */
+    private static final long NANOS_IN_SECOND = 1000L * 1000L * 1000L;
+
+    /**
+     * The time to sleep/yield until the next frame
+     */
+    private static long nextFrame = 0;
+
+    /**
+     * whether the initialisation code has run
+     */
+    private static boolean initialised = false;
+
+    /**
+     * for calculating the averages the previous sleep/yield times are stored
+     */
+    private static RunningAvg sleepDurations = new RunningAvg(10);
+    private static RunningAvg yieldDurations = new RunningAvg(10);
+
+
+    /**
+     * An accurate sync method that will attempt to run at a constant frame rate. It should be called once every frame.
+     *
+     * @param fps - the desired frame rate, in frames per second
+     */
+    public static void sync(int fps) {
+        if (fps <= 0) return;
```

Comment:
Aiieeee brackets here and the next line. What does this class do anyway? Is it copied in from somewhere or original?